go.mod

```diff
@@ -68,6 +68,7 @@ require (
 	github.com/evanphx/json-patch v5.6.0+incompatible
 	github.com/golang/mock v1.6.0
 	github.com/portworx/kdmp v0.4.1-0.20240730181245-423ce1fdc45b
+	go.uber.org/mock v0.4.0
```

Comment:
I have to check why this is required. I will get back to you.

Reply:
Looks like with current mockgen multilevel interface mocking is not possible.
```
✗ mockgen -source=vendor/github.com/portworx/sched-ops/k8s/externalstorage/externalstorage.go -destination=pkg/snapshot/controllers/snapshot.mock.go
2024/08/30 10:05:52 Loading input failed: vendor/github.com/portworx/sched-ops/k8s/externalstorage/externalstorage.go:21:2: unknown embedded interface SnapshotOps
```

Reply:
Removed the mocking logic for snapshotschedule. Using mocking only for snapshots now.

Reply:
@adityadani , we are using the above package for mocking multilevel interfaces.

Reply:
It uses Apache license 2.0 which allows private use and commercial usage as well. It can be found here: https://github.com/uber-go/mock/blob/main/LICENSE